projects/addon-mobile/directives/ripple/ripple.directive.ts

```diff
@@ -1,79 +1,158 @@
-import {Directive, ElementRef, inject, Input, Renderer2} from '@angular/core';
+import {
+    DestroyRef,
+    Directive,
+    HostListener,
+    inject,
+    Input,
+    Renderer2,
+} from '@angular/core';
 import {takeUntilDestroyed} from '@angular/core/rxjs-interop';
 import {
     TUI_FALSE_HANDLER,
     TUI_TRUE_HANDLER,
+    tuiIsString,
+    tuiPx,
     tuiTypedFromEvent,
     tuiWithStyles,
 } from '@taiga-ui/cdk';
-import {map, mergeMap, race, switchMap, take, takeUntil, tap, timer} from 'rxjs';
-
+import {TUI_ANIMATIONS_SPEED, tuiGetDuration} from '@taiga-ui/core';
 import {
-    RIPPLE_OFF,
-    TUI_RIPPLE_END,
-    TUI_RIPPLE_PROVIDERS,
-    TUI_RIPPLE_START,
-} from './ripple.providers';
-import {TuiRippleStylesComponent} from './ripple-styles.component';
+    defaultIfEmpty,
+    map,
+    merge,
+    race,
+    switchMap,
+    take,
+    takeUntil,
+    tap,
+    timer,
+} from 'rxjs';
+
+import {TuiRippleComponent} from './ripple.component';
 
 const TOUCH_MOVE_DELAY = 100;
+const TOUCH_MOVE_DELAY_TIMER = timer(TOUCH_MOVE_DELAY).pipe(map(TUI_FALSE_HANDLER));
+const RIPPLE_ON_ANIMATION: Keyframe[] = [
+    {
+        transform: 'scale(0)',
+        opacity: '0.12',
+    },
+    {
+        opacity: '0.12',
+    },
+];
+const RIPPLE_OFF_ANIMATION: Keyframe[] = [
+    {
+        opacity: '0.12',
+    },
+    {
+        opacity: '0',
+    },
+];
+
+function selectorScopeFallback(value: string): string {
+    return tuiIsString(value) && value ? value : ':scope';
+}
 
-@Directive({
-    selector: '[tuiRipple]',
-    providers: TUI_RIPPLE_PROVIDERS,
-})
+@Directive({standalone: true, selector: '[tuiRipple]'})
 export class TuiRippleDirective {
-    private readonly el: HTMLElement = inject(ElementRef).nativeElement;
+    private readonly destroy$ = inject(DestroyRef);
     private readonly renderer = inject(Renderer2);
-    private readonly start$ = inject(TUI_RIPPLE_START);
-    private readonly end$ = inject(TUI_RIPPLE_END);
+    private readonly animationSpeed = inject(TUI_ANIMATIONS_SPEED);
 
-    @Input()
-    public tuiRipple?: string | '';
-
-    protected readonly nothing = tuiWithStyles(TuiRippleStylesComponent);
+    @Input({
+        alias: 'tuiRipple',
+        transform: selectorScopeFallback,
+    })
+    public selector = ':scope';
 
     constructor() {
-        const touchEnd$ = tuiTypedFromEvent(this.el, 'touchend');
-        const touchMove$ = tuiTypedFromEvent(this.el, 'touchmove');
+        tuiWithStyles(TuiRippleComponent);
+    }
+
+    @HostListener('pointerdown.silent', [
+        '$event.clientX',
+        '$event.clientY',
+        '$event.target',
+    ])
+    protected start(clientX: number, clientY: number, target: HTMLElement): void {
```

Comment:
nitpicking: you can name those just `x` and `y` for brevity, since you already destructured event above. This way line 85 will not span 4 lines and we will not lose readability.